src/components/ScriptExplorer/SignatureImporter.jsx

```diff
@@ -304,72 +314,159 @@ class SignatureImporter extends React.Component {
       errback("Not enough signatures (must be exactly one for each input).");
       return;
     }
-    if (inputsSignatures.length > inputs.length) {
-      errback("Too many signatures (must be exactly one for each input).");
-      return;
+
+    if (inputsSignatures.length % inputs.length !== 0) {
+      errback(
+        "Not the right number of signatures (must be exactly one for each input)."
```

Comment:
"Number of signatures must be a multiple of number of inputs."

Reply:
Also add a `return` to short-circuit out of this function after calling `errback`